lua/starfall/sflib.lua

```diff
@@ -1096,7 +1096,7 @@ do
 		local tableLookup = {}
 
 		local function typeToString(val)
-			local func = typetostringfuncs[TypeID(val)]
+			local func = typetostringfuncs[IsColor(val) and TYPE_COLOR or TypeID(val)]
```

Comment:
Can you instead do this inside `typetostringfuncs[TYPE_TABLE] = function(ss, val)` (line 1103)

`if IsColor(val) then return typetostringfuncs[TYPE_COLOR](ss, val) end`